code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -303,6 +317,19 @@
 		togglelock(user)
 		return
 
+/obj/structure/closet/attack_robot(mob/user)
+	if(user.Adjacent(src))
+		return attack_hand(user)
+	else
+		return attack_ai(user)
+
+/obj/structure/closet/attack_ai(mob/user)
+	if(secure && opened)
+		user << "<span class='warning'>You cannot lock open locker!</span>"

```

Comment:
How about "You cannot lock this while it is open", or "[src] must be closed before you can lock it!"?